vmdb/app/models/tag.rb

```diff
@@ -104,4 +107,29 @@ def self.filter_ns(tags, ns)
   def ==(comparison_object)
     super || name.downcase == comparison_object.to_s.downcase
   end
+
+  def category
+    @category ||= Classification.find_by_name(name_path.split('/').first, nil)
+  end
+
+  def show
+    category && category.show
+  end
+
+  def categorization
+    @categorization ||= begin
+      !show ? {} : {
+        "name"         => classification.name,
+        "description"  => classification.description,
+        "category"     => {"name" => category.name, "description" => category.description},
+        "display_name" => "#{category.description}: #{classification.description}"
+      }
+    end
+  end
+
+  private
+
+  def name_path
+    @name_path ||= name.dup.sub!(%r{/[^/]*/}, "")
```

Comment:
no need for dup... `name.sub(%r{/[^/]*/}, "")`

Also, this should be locked to the start of the string with `^`